presto-cli/src/main/java/com/facebook/presto/cli/Console.java

```diff
@@ -92,7 +92,7 @@ public void run()
         boolean hasQuery = !Strings.isNullOrEmpty(clientOptions.execute);
         boolean isFromFile = !Strings.isNullOrEmpty(clientOptions.file);
 
-        if (!hasQuery || !isFromFile) {
+        if (!hasQuery && !isFromFile) {
```

Comment:
It turns out this is backwards. The purpose of `AnsiOutputStream` is to filter out ANSI escape sequences when not using a terminal or to interpret them differently using a subclass like `WindowsAnsiOutputStream`.